### PR TITLE
Feature/implement the grant tokenizer role and test 75

### LINKED ISSUE #75: Implement the `grant_tokenizer_role` and test
**Goal**  
Let an admin grant the `TOKENIZER_ROLE` to a specified account.

---

### Implementation Details

```cairo
fn grant_tokenizer_role(
    ref self: ContractState,
    account: ContractAddress
)
```

- Call `self.accesscontrol.grant_role(TOKENIZER_ROLE, account);`
- Emit `TokenizerRoleGranted` event.

---

### Acceptance Criteria

- [ ] `account` receives the `TOKENIZER_ROLE`.
- [ ] Event is emitted with `account` and `granter`.
- [ ] Write a test for success and failure case.

---

### Suggested Tests

- Admin can grant role.
- Non-admin cannot grant role.
- Role check confirms new assignment.

### PR DESCRIPTION
**Pull request by anihdev**
## Description 
This PR implements the `grant_tokenizer_role` functionality and comprehensive test coverage as specified in #75. The implementation allows admins to grant the `TOKENIZER_ROLE` to specified accounts with proper access control and event emission.

Restructured the contrract and adjusted the `lib.cairo` file.

## Key improvements include:

- **Access Control Implementation**: Properly configured role hierarchy with `DEFAULT_ADMIN_ROLE` managing `TOKENIZER_ROLE`
- **Constructor Enhancement**: Fixed component initialization order and role setup for consistent access control
- **Comprehensive Test Coverage**: Added thorough test suite covering success, failure, and edge cases
- **Event Validation**: Implemented proper event emission with `TokenizerRoleGranted` event
- **Address Consistency**: Fixed test setup to use consistent addresses throughout

### Core Function
```cairo
fn grant_tokenizer_role(ref self: ContractState, account: ContractAddress) {
   // Check that caller has DEFAULT_ADMIN_ROLE
   self.accesscontrol.assert_only_role(DEFAULT_ADMIN_ROLE);
   
   // Grant the TOKENIZER_ROLE to the specified account
   self.accesscontrol._grant_role(TOKENIZER_ROLE, account);
   
   // Emit the TokenizerRoleGranted event
   let granter = get_caller_address();
   self.emit(TokenizerRoleGranted { account, granter });
}
```

## Tests Added
* test_grant_tokenizer_role_success - Verifies successful role granting by admin
* test_grant_tokenizer_role_non_admin_fails - Ensures non-admin access is blocked
* test_has_tokenizer_role_after_grant - Confirms role assignment via interface
* test_multiple_tokenizer_roles - Tests granting roles to multiple accounts
* test_admin_role_check - Validates proper role setup in constructor
* test_grant_role_to_same_account_twice - Handles duplicate role grants gracefully

## Related Issues 
Fixes #75  - Implement the grant_tokenizer_role and test

## Changes Made 
[✅] ✨ Feature Implementation
[✅] 🐛 Bug Fix
[✅] 📚 Documentation Update
[✅] 🔨 Refactoring

## Acceptance Criteria Met 

[✅] account receives the TOKENIZER_ROLE
[✅] Event is emitted with account and granter
[✅] Admin can grant role
[✅] Non-admin cannot grant role
[✅] Role check confirms new assignment
[✅] Additional edge cases covered

### CODE REVIEW

**Comment by GideonBature** [comment]
@anihdev please what's the update on this?

**Comment by anihdev** [comment]
@GideonBature apologies for the delay boss please check now

**Comment by anihdev** [comment]
It’s alright

**Comment by anihdev** [comment]
i had no intention of stressing this, i was trying to better organise the contract and add missing imports, but it seems ur updated commit did that without my knowledge, check now for any issue @GideonBature

File: src/lib.cairo [hunk 1]
```diff
@@ -5,6 +5,8 @@ pub mod interfaces {
 pub mod events {
```
Comment (GideonBature) [inline src/lib.cairo:5]: please don't touch the `lib.cairo` file, everything that needs to be there is there already chief....

File: src/contracts/rwa_factory.cairo [hunk 2]
```diff
@@ -43,6 +43,10 @@ mod RWAFactory {
         AccessControlComponent::AccessControlImpl<ContractState>;
     impl AccessControlInternalImpl = AccessControlComponent::InternalImpl<ContractState>;
 
+    // Implement SRC5 mixin
```
Comment (GideonBature) [inline src/contracts/rwa_factory.cairo:46]: I believe this one is there already too. So can you please take if off?

**Comment by GideonBature** [comment]
Thank you brother. This is sharp!!!
